src/vm/wks/CMakeLists.txt

```diff
@@ -1,67 +1,115 @@
+
 if (WIN32)
   add_precompiled_header(common.h ../common.cpp VM_SOURCES_WKS)
   # mscorlib.cpp does not compile with precompiled header file
   set_source_files_properties(../mscorlib.cpp PROPERTIES COMPILE_FLAGS "/Y-") 
   # .c file cannot use cpp precompiled header 
   set_source_files_properties(../microsoft.comservices_i.c PROPERTIES COMPILE_FLAGS "/Y-") 
 
-  # asm files require preprocessing using cl.exe on arm64
+  # asm files require preprocessing using cl.exe on arm32 and arm64
   if(CLR_CMAKE_PLATFORM_ARCH_ARM64)
+
       foreach(ASM_FILE ${VM_SOURCES_WKS_ARCH_ASM})
+
+          # Preprocess each asm source file
           get_filename_component(name ${ASM_FILE} NAME_WE)
           set(ASM_PREPROCESSED_FILE ${CMAKE_CURRENT_BINARY_DIR}/${name}.asm)
           preprocess_def_file(${ASM_FILE} ${ASM_PREPROCESSED_FILE})
+
           set(VM_SOURCES_WKS_ARM64_PREPROCESSED_ASM ${VM_SOURCES_WKS_ARM64_PREPROCESSED_ASM} ${ASM_PREPROCESSED_FILE})
+
       endforeach()
 
       set(VM_SOURCES_WKS_ARCH_ASM ${VM_SOURCES_WKS_ARM64_PREPROCESSED_ASM})
+
+  elseif(CLR_CMAKE_PLATFORM_ARCH_ARM)
+
+      get_include_directories_asm(ASM_INCLUDE_DIRECTORIES)
+
+      foreach(ASM_FILE ${VM_SOURCES_WKS_ARCH_ASM})
+
+          # Inserts a custom command in CMake build to preprocess each asm source file
+          get_filename_component(name ${ASM_FILE} NAME_WE)
+          file(TO_CMAKE_PATH "${CMAKE_CURRENT_BINARY_DIR}/${name}.asm" ASM_PREPROCESSED_FILE)
+          preprocess_def_file(${ASM_FILE} ${ASM_PREPROCESSED_FILE})
+
+          # On Arm32, compile the preprocessed binary to .obj
+          # We do not pass any defines since we have already done pre-processing above
+          set (ASM_CMDLINE "-o ${CMAKE_CURRENT_BINARY_DIR}/${name}.obj ${ASM_PREPROCESSED_FILE}")
+
+          # Generate the batch file that will invoke the assembler
+          file(TO_CMAKE_PATH "${CMAKE_CURRENT_BINARY_DIR}/runasm_${name}_${CMAKE_BUILD_TYPE}.cmd" ASM_SCRIPT_FILE)
```

Comment:
Not doing TO_CMAKE_PATH didnt work for me due to spaces in path. I believe the difference between Arm32 and Arm64 build is that, in Arm64, the assembler is just found while in Arm32, we set path to it.